homeassistant/components/subaru/config_flow.py

```diff
@@ -0,0 +1,168 @@
+    """Return a set of configured Subaru instances."""
+    return {entry.title for entry in hass.config_entries.async_entries(DOMAIN)}
+
+
+class SubaruConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Subaru."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_CLOUD_POLL
+
+    async def async_step_user(self, user_input=None):
+        """Handle the start of the config flow."""
+        error = None
+
+        if not user_input:
+            return self.async_show_form(
+                step_id="user",
+                data_schema=vol.Schema(
+                    {
+                        vol.Required(CONF_USERNAME): str,
+                        vol.Required(CONF_PASSWORD): str,
+                        vol.Required(CONF_PIN): str,
+                    }
+                ),
+            )
+
+        if user_input[CONF_USERNAME] in configured_instances(self.hass):
+            return self.async_abort(reason="already_configured")
+
+        try:
+            info = await validate_input(self.hass, user_input)
+        except InvalidCredentials:
+            error = {"base": "invalid_auth"}
+        except InvalidPIN:
+            error = {"base": "invalid_pin"}
+        except SubaruException as ex:
+            _LOGGER.error("Unable to communicate with Subaru API: %s", ex.message)
+            return self.async_abort(reason="cannot_connect")
+
+        if error:
+            return self.async_show_form(
+                step_id="user",
+                data_schema=vol.Schema(
+                    {
+                        vol.Required(
+                            CONF_USERNAME, default=user_input.get(CONF_USERNAME)
+                        ): str,
+                        vol.Required(
+                            CONF_PASSWORD, default=user_input.get(CONF_PASSWORD)
+                        ): str,
+                        vol.Required(CONF_PIN, default=user_input.get(CONF_PIN)): str,
+                    }
+                ),
+                errors=error,
+            )
+
+        return self.async_create_entry(title=user_input[CONF_USERNAME], data=info)
+
+    @staticmethod
+    @callback
+    def async_get_options_flow(config_entry):
+        """Get the options flow for this handler."""
+        return OptionsFlowHandler(config_entry)
+
+
+class OptionsFlowHandler(config_entries.OptionsFlow):
+    """Handle a option flow for Subaru."""
+
+    def __init__(self, config_entry: config_entries.ConfigEntry):
+        """Initialize options flow."""
+        self.config_entry = config_entry
+
+    async def async_step_init(self, user_input=None):
+        """Handle options flow."""
+        if user_input is not None:
+            return self.async_create_entry(title="", data=user_input)
+
+        data_schema = vol.Schema(
+            {
+                vol.Optional(
+                    CONF_SCAN_INTERVAL,
+                    default=self.config_entry.options.get(
```

Comment:
If you're setting a default this isn't really optional :)

Reply:
Change made.